src/components/lib/crt/crt.c

```diff
@@ -167,6 +217,74 @@ crt_comp_create_with(struct crt_comp *c, char *name, compid_t id, struct crt_com
 	cos_compinfo_init(cos_compinfo_get(c->comp_res),
 			  r->ptc, r->ctc, r->compc, r->heap_ptr, r->captbl_frontier,
 			  cos_compinfo_get(cos_defcompinfo_curr_get()));
+	return 0;
+}
+
+/** 
+ * Create the component from the checkpoint
+ * c is the new component
+ * chkpt is the checkpoint used to create c
+ * name is c's name
```

Comment:
This comment doesn't follow the structure of this type of comment. See `crt_comp_create` below. Just copy and paste that comment and change the sections. Use that style of argument documentation, use of empty lines, etc...